apps/foundation-demo/src/app/product/components/add-to-cart/add-to-cart.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, Output, EventEmitter } from '@angular/core';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'add-to-cart',
+  templateUrl: './add-to-cart.component.html',
+  styleUrls: ['./add-to-cart.component.scss']
+})
+export class AddToCartComponent {
+
+  @Output() addToCart: EventEmitter<any> = new EventEmitter();
+
+  constructor(
+    private router: Router
+  ) { }
+
+  onAddToCart() {
+    this.emitAddToCart();
+    this.redirectToCart();
+  }
+
+  emitAddToCart() {
+    this.addToCart.emit();
+  }
+
+  redirectToCart() {
+    this.router.navigateByUrl('/cart');
```

Comment:
Move this to an effect. This is a clear instance of a side-effect that shouldn't be known about by this component.

Reply:
Should this effect be in the foundation-demo? Or daffodil/core? At first I thought daffodil/core, but daffodil/core currently doesn't know anything about any urls or navigation, so we'd be defining what the user's urls need to be, even though they'll be the ones writing out the routing. I guess it could go in foundation-demo, but that seemed weird to me. Can't really think of a good reason that it shouldn't go in foundation-demo though, since we won't really be adding extra dependencies. So it should be added to foundation-demo?

Reply:
I'd argue that effect should be in foundation-demo. This should be a hook into the AddToCart success action.

Reply:
done